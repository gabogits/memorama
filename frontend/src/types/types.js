export const types = {
  LOGIN: "LOGIN",
  LOGOUT: "LOGOUT",
  SET_CARDS_GAME: "SET_CARDS_GAME",
  RESET_CATEGORY: "RESET_CATEGORY",
  RESET: "RESET",
};
