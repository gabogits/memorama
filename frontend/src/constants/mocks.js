export const arrayImages = [
  {
    parid: "2JcixB1Ky3I",
    url: "https://images.unsplash.com/photo-1570824104453-508955ab713e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "46TvM-BVrRI",
    url: "https://images.unsplash.com/photo-1589883661923-6476cb0ae9f2?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwyfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "YCPkW_r_6uA",
    url: "https://images.unsplash.com/photo-1517331156700-3c241d2b4d83?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwzfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "mBRfYA0dYYE",
    url: "https://images.unsplash.com/photo-1583795128727-6ec3642408f8?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw0fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "uy5t-CJuIK4",
    url: "https://images.unsplash.com/photo-1541781774459-bb2af2f05b55?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw1fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "FbhNdD1ow2g",
    url: "https://images.unsplash.com/photo-1557246565-8a3d3ab5d7f6?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw2fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "MNju0A6EeE0",
    url: "https://images.unsplash.com/photo-1559235038-1b0fadf76f78?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw3fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "gSSK4u8yPpM",
    url: "https://images.unsplash.com/photo-1568043210943-0e8aac4b9734?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw4fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "qs4j-39TaBQ",
    url: "https://images.unsplash.com/photo-1595511890410-3b8dc237a537?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw5fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
  },
  {
    parid: "21NRDbMJF94",
    url: "https://images.unsplash.com/photo-1574063413132-354db9f190fd?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxMHx8Y2F0c3xlbnwwfDB8MXx8MTY0MzgyNTAxNA&ixlib=rb-1.2.1&q=80&w=400",
  },
];
export const mockGameCards = [
  {
    parid: "gSSK4u8yPpM",
    url: "https://images.unsplash.com/photo-1568043210943-0e8aac4b9734?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw4fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "UanozUmNaQ",
    flipped: false,
    active: true,
  },
  {
    parid: "2JcixB1Ky3I",
    url: "https://images.unsplash.com/photo-1570824104453-508955ab713e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "HSmfLRJf4b",
    flipped: false,
    active: true,
  },
  {
    parid: "uy5t-CJuIK4",
    url: "https://images.unsplash.com/photo-1541781774459-bb2af2f05b55?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw1fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "feRFoRKQ6Z",
    flipped: false,
    active: true,
  },
  {
    parid: "mBRfYA0dYYE",
    url: "https://images.unsplash.com/photo-1583795128727-6ec3642408f8?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw0fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "R6-xYjqtkH",
    flipped: false,
    active: true,
  },
  {
    parid: "FbhNdD1ow2g",
    url: "https://images.unsplash.com/photo-1557246565-8a3d3ab5d7f6?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw2fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "9F7o3fBP29",
    flipped: false,
    active: true,
  },
  {
    parid: "MNju0A6EeE0",
    url: "https://images.unsplash.com/photo-1559235038-1b0fadf76f78?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw3fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "F6jx1v-Vyd",
    flipped: false,
    active: true,
  },
  {
    parid: "gSSK4u8yPpM",
    url: "https://images.unsplash.com/photo-1568043210943-0e8aac4b9734?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw4fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "oLzJktU-Az",
    flipped: false,
    active: true,
  },
  {
    parid: "FbhNdD1ow2g",
    url: "https://images.unsplash.com/photo-1557246565-8a3d3ab5d7f6?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw2fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "P9hCUD77_p",
    flipped: false,
    active: true,
  },
  {
    parid: "YCPkW_r_6uA",
    url: "https://images.unsplash.com/photo-1517331156700-3c241d2b4d83?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwzfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "MUa5PByAj8",
    flipped: false,
    active: true,
  },
  {
    parid: "YCPkW_r_6uA",
    url: "https://images.unsplash.com/photo-1517331156700-3c241d2b4d83?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwzfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "1zTBC5WcO-",
    flipped: false,
    active: true,
  },
  {
    parid: "mBRfYA0dYYE",
    url: "https://images.unsplash.com/photo-1583795128727-6ec3642408f8?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw0fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "agIioCJ5_t",
    flipped: false,
    active: true,
  },
  {
    parid: "uy5t-CJuIK4",
    url: "https://images.unsplash.com/photo-1541781774459-bb2af2f05b55?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw1fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "V02rFiPXi_",
    flipped: false,
    active: true,
  },
  {
    parid: "21NRDbMJF94",
    url: "https://images.unsplash.com/photo-1574063413132-354db9f190fd?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxMHx8Y2F0c3xlbnwwfDB8MXx8MTY0MzgyNTAxNA&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "4qzwACkR6F",
    flipped: false,
    active: true,
  },
  {
    parid: "qs4j-39TaBQ",
    url: "https://images.unsplash.com/photo-1595511890410-3b8dc237a537?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw5fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "lWOOBcAgTf",
    flipped: false,
    active: true,
  },
  {
    parid: "MNju0A6EeE0",
    url: "https://images.unsplash.com/photo-1559235038-1b0fadf76f78?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw3fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "vJG_OfRr0p",
    flipped: false,
    active: true,
  },
  {
    parid: "46TvM-BVrRI",
    url: "https://images.unsplash.com/photo-1589883661923-6476cb0ae9f2?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwyfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "70ZYeFj5sc",
    flipped: false,
    active: true,
  },
  {
    parid: "46TvM-BVrRI",
    url: "https://images.unsplash.com/photo-1589883661923-6476cb0ae9f2?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwyfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "HTbumxhoHL",
    flipped: false,
    active: true,
  },
  {
    parid: "21NRDbMJF94",
    url: "https://images.unsplash.com/photo-1574063413132-354db9f190fd?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxMHx8Y2F0c3xlbnwwfDB8MXx8MTY0MzgyNTAxNA&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "sD2Y-y1-jT",
    flipped: false,
    active: true,
  },
  {
    parid: "2JcixB1Ky3I",
    url: "https://images.unsplash.com/photo-1570824104453-508955ab713e?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHwxfHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "fctB2ZTC44",
    flipped: false,
    active: true,
  },
  {
    parid: "qs4j-39TaBQ",
    url: "https://images.unsplash.com/photo-1595511890410-3b8dc237a537?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwyMDM2Nzh8MHwxfHNlYXJjaHw5fHxjYXRzfGVufDB8MHwxfHwxNjQzODI1MDE0&ixlib=rb-1.2.1&q=80&w=400",
    nanoId: "krA7fPHREn",
    flipped: false,
    active: true,
  },
];
export const players = {
  numberPlayers: 2,
  dataPlayers: [
    { avatar: "witch", id: 1, name: "juan" },
    { avatar: "dracula", id: 2, name: "bety" },
  ],
};
