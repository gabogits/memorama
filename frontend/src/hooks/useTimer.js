const { useState, useRef, useEffect } = require("react");

const useTimer = (initialState = 0) => {
  const isMounted = useRef(true);
  const [timer, setTimer] = useState(initialState);
  const [isActive, setIsActive] = useState(false);
  const [isPaused, setIsPaused] = useState(false);
  const countRef = useRef(null);

  const handleStart = () => {
    setIsActive(true);
    setIsPaused(true);

    countRef.current = setInterval(() => {
      if (isMounted.current) {
        setTimer((timer) => timer + 1);
      }
    }, 1000);
  };
  const handlePause = () => {
    clearInterval(countRef.current);
    setIsPaused(false);
  };

  const handleResume = () => {
    setIsPaused(true);
    countRef.current = setInterval(() => {
      if (isMounted.current) {
        setTimer((timer) => timer + 1);
      }
    }, 1000);
  };

  const handleReset = () => {
    clearInterval(countRef.current);
    setIsActive(false);
    setIsPaused(false);
    setTimer(0);
  };
  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);
  return {
    timer,
    isActive,
    isPaused,
    handleStart,
    handlePause,
    handleResume,
    handleReset,
  };
};
export default useTimer;
