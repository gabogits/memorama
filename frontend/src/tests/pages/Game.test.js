import React from "react";
import { mount } from "enzyme";
import { Router } from "react-router-dom";
import GameContext from "../../context/GameContext";
import "@testing-library/jest-dom";
import { types } from "../../types/types";
import Game from "../../pages/Game";
import { useGame } from "../../hooks/useGame";
import useTimer from "../../hooks/useTimer";
import { arrayImages, mockGameCards } from "../../constants/mocks";
jest.mock("../../hooks/useGame");
jest.mock("../../hooks/useTimer");

describe("Pruebas en <Game />", () => {
  let historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: jest.fn(),
    listen: jest.fn(),
    createHref: jest.fn(),
  };
  const contextValue = {
    dispatch: jest.fn(),
    state: {
      numberPlayers: 2,
      dataPlayers: [
        { avatar: "witch", id: 1, name: "juan" },
        { avatar: "dracula", id: 2, name: "bety" },
      ],
      arrayImages,
      category: "cats",
    },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("deberia redirigirlo a tema si el usuario no cuenta con array de imagenes", async () => {
    useGame.mockReturnValue({
      playerMove: { avatar: "witch", id: 1, name: "juan" },
      cardsGame: [],
      gameData: [
        { avatar: "witch", id: 1, name: "juan", score: 0 },
        { avatar: "dracula", id: 2, name: "bety", score: 0 },
      ],
      endGame: false,
      score: null,
      initGame: jest.fn(),
      handleCardFlip: jest.fn(),
    });
    useTimer.mockReturnValue({
      timer: 0,
      isActive: false,
      isPaused: false,
      handleStart: jest.fn(),
      handlePause: jest.fn(),
      handleResume: jest.fn(),
      handleReset: jest.fn(),
    });
    const contextValueNoImages = {
      dispatch: jest.fn(),
      state: {
        numberPlayers: 2,
        dataPlayers: [
          { avatar: "witch", id: 1, name: "juan" },
          { avatar: "dracula", id: 2, name: "bety" },
        ],
        arrayImages: [],
        category: "cats",
      },
    };
    mount(
      <GameContext.Provider value={contextValueNoImages}>
        <Router history={historyMock}>
          <Game />
        </Router>
      </GameContext.Provider>
    );
    expect(historyMock.push).toHaveBeenCalledWith("/tema");
  });
  test("Debe de resetarse el state al dar clic en endSesionGame ademas de redirigirte a /", async () => {
    useGame.mockReturnValue({
      playerMove: { avatar: "witch", id: 1, name: "juan" },
      cardsGame: [],
      gameData: [
        { avatar: "witch", id: 1, name: "juan", score: 0 },
        { avatar: "dracula", id: 2, name: "bety", score: 0 },
      ],
      endGame: false,
      score: null,
      initGame: jest.fn(),
      handleCardFlip: jest.fn(),
    });
    useTimer.mockReturnValue({
      timer: 0,
      isActive: false,
      isPaused: false,
      handleStart: jest.fn(),
      handlePause: jest.fn(),
      handleResume: jest.fn(),
      handleReset: jest.fn(),
    });

    const wrapper = mount(
      <GameContext.Provider value={contextValue}>
        <Router history={historyMock}>
          <Game />
        </Router>
      </GameContext.Provider>
    );
    wrapper.find(".btn-icon-2.icon-endGame").simulate("click");

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.RESET,
      payload: {
        numberPlayers: 0,
        dataPlayers: [],
        arrayImages: [],
        category: null,
      },
    });

    expect(historyMock.push).toHaveBeenCalledWith("/");
  });

  test("Debe de reiniciarte el juego con los valores por defecto", async () => {
    useGame.mockReturnValue({
      playerMove: { avatar: "witch", id: 1, name: "juan" },
      cardsGame: mockGameCards,
      gameData: [
        { avatar: "witch", id: 1, name: "juan", score: 0 },
        { avatar: "dracula", id: 2, name: "bety", score: 0 },
      ],
      endGame: false,
      score: null,
      initGame: jest.fn(),
      handleCardFlip: jest.fn(),
    });
    useTimer.mockReturnValue({
      timer: 0,
      isActive: false,
      isPaused: false,
      handleStart: jest.fn(),
      handlePause: jest.fn(),
      handleResume: jest.fn(),
      handleReset: jest.fn(),
    });

    const wrapper = mount(
      <GameContext.Provider value={contextValue}>
        <Router history={historyMock}>
          <Game />
        </Router>
      </GameContext.Provider>
    );
    wrapper.find(".btn-icon-2.icon-repeat").simulate("click");

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(".flipped").length).toBe(0);
    expect(wrapper.find(".disabled").length).toBe(0);
    expect(wrapper.find(".cards-list").exists()).toBe(true);
    expect(wrapper.find(".modal-box").hasClass("active")).toBe(false);
    expect(wrapper.find(".screen").hasClass("shadow")).toBe(false);
    expect(wrapper.find(".options-aftergame").exists()).toBe(false);
  });

  test("Debe de mostrar la informacion de los jugadores", async () => {
    useGame.mockReturnValue({
      playerMove: { avatar: "witch", id: 1, name: "juan" },
      cardsGame: mockGameCards,
      gameData: [
        { avatar: "witch", id: 1, name: "juan", score: 0 },
        { avatar: "dracula", id: 2, name: "bety", score: 0 },
      ],
      endGame: false,
      score: null,
      initGame: jest.fn(),
      handleCardFlip: jest.fn(),
    });
    useTimer.mockReturnValue({
      timer: 0,
      isActive: false,
      isPaused: false,
      handleStart: jest.fn(),
      handlePause: jest.fn(),
      handleResume: jest.fn(),
      handleReset: jest.fn(),
    });

    const wrapper = mount(
      <GameContext.Provider value={contextValue}>
        <Router history={historyMock}>
          <Game />
        </Router>
      </GameContext.Provider>
    );

    expect(wrapper.find(".header-game .header-gamedata-item").length).toBe(2);

    expect(
      wrapper
        .find(".header-game .header-gamedata-item")
        .at(0)
        .find(".header-gamedata-data")
        .text()
        .trim()
    ).toBe("juan 0");
    expect(
      wrapper
        .find(".header-game .header-gamedata-item")
        .at(1)
        .find(".header-gamedata-data")
        .text()
        .trim()
    ).toBe("bety 0");

    expect(wrapper.find(".timer p").text().trim()).toBe("00:00:00");
  });

  test("Debe de mostrar el modal cuando termine el juego, ocultarse las tarjetas, mostrar las opciones repetir o terminar juego", async () => {
    const contextValueEndGame = {
      dispatch: jest.fn(),
      state: {
        numberPlayers: 1,
        dataPlayers: [{ avatar: "witch", id: 1, name: "juan" }],
        arrayImages,
        category: "cats",
      },
    };
    useGame.mockReturnValue({
      playerMove: { avatar: "witch", id: 1, name: "juan" },
      cardsGame: mockGameCards,
      gameData: [{ avatar: "witch", id: 1, name: "juan", score: 10 }],
      endGame: true,
      score: 2332,
      initGame: jest.fn(),
      handleCardFlip: jest.fn(),
    });
    useTimer.mockReturnValue({
      timer: 90,
      isActive: false,
      isPaused: false,
      handleStart: jest.fn(),
      handlePause: jest.fn(),
      handleResume: jest.fn(),
      handleReset: jest.fn(),
    });

    const wrapper = mount(
      <GameContext.Provider value={contextValueEndGame}>
        <Router history={historyMock}>
          <Game />
        </Router>
      </GameContext.Provider>
    );
    expect(wrapper.find(".cards-list").exists()).toBe(false);
    expect(wrapper.find(".modal-box").hasClass("active")).toBe(true);
    expect(wrapper.find(".screen").hasClass("shadow")).toBe(true);
    expect(wrapper.find(".timer p").text().trim()).toBe("00:01:30");
    expect(wrapper.find(".modal-box .score-modal").text().trim()).toBe("2332");
    expect(wrapper.find(".options-aftergame").exists()).toBe(true);
  });
});
