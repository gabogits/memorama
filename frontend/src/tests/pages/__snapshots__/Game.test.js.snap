// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Game /> Debe de reiniciarte el juego con los valores por defecto 1`] = `
<main>
  <header>
    <div
      className=" container container-header"
    >
      <div
        className="logo"
      >
        <figure>
          <a
            onClick={[Function]}
          >
            <span
              className="logo-desk"
            >
              MEMORAMA
            </span>
            <span
              className="logo-mobile"
            >
              M
            </span>
             
          </a>
        </figure>
      </div>
      <nav>
        <div
          className="nav-left"
        />
        <div
          className="nav-right"
        >
          <div
            className="nav-list"
          >
            <ul>
              <li>
                <button
                  onClick={[Function]}
                >
                  Inicio
                </button>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Tema
                </a>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Juego
                </a>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Ranking
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </header>
  <div
    className="header-game"
  >
    <div
      className="header-game-container"
    >
      <div
        className="header-gamedata"
      >
        <div
          className="header-gamedata-item"
          key="1"
        >
          <div
            className="header-gamedata-avatar active"
          >
            <img
              alt="avatar"
              src="witch.png"
            />
          </div>
          <div
            className="header-gamedata-data"
          >
            juan
            <br />
            <strong>
               
              0
            </strong>
          </div>
        </div>
        <div
          className="header-gamedata-item"
          key="2"
        >
          <div
            className="header-gamedata-avatar "
          >
            <img
              alt="avatar"
              src="dracula.png"
            />
          </div>
          <div
            className="header-gamedata-data"
          >
            bety
            <br />
            <strong>
               
              0
            </strong>
          </div>
        </div>
      </div>
      <div
        className="timer-buttons"
      >
        <div
          className="timer-container"
        >
          <div
            className="timer"
          >
            <p>
              00:00:00
            </p>
          </div>
        </div>
        <div
          className="container-icon-menu"
        >
          <button
            className="btn-icon-2 icon-repeat"
            onClick={[Function]}
            type="submit"
            value="enviar"
          />
          <button
            className="btn-icon-2 icon-endGame"
            onClick={[Function]}
            type="submit"
            value="enviar"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="cards-board"
  >
    <div
      className="cards-list"
    >
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="flip-card  "
        onClick={[Function]}
      >
        <div
          className="flip-card-inner"
        >
          <div
            className="flip-card-front"
          />
          <div
            className="flip-card-back "
          >
            <img
              alt="Avatar"
              src="idontknow.jpg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  Array [
    <div
      className="modal-box medium-size "
    >
      <div
        className="head-modal"
      >
        <h2>
          Score final
        </h2>
      </div>
      <div
        className="modal-box-inner"
      >
        <div
          className="header-gamedata center-jus"
        >
          <div
            className="header-gamedata-item"
            key="1"
          >
            <div
              className="header-gamedata-avatar"
            >
              <img
                alt="avatar"
                src="witch.png"
              />
            </div>
            <div
              className="header-gamedata-data"
            >
              <span>
                 
                juan
              </span>
               
              <br />
              <strong
                className="score-modal "
              >
                 
                0
              </strong>
            </div>
          </div>
          <div
            className="header-gamedata-item"
            key="2"
          >
            <div
              className="header-gamedata-avatar"
            >
              <img
                alt="avatar"
                src="dracula.png"
              />
            </div>
            <div
              className="header-gamedata-data"
            >
              <span>
                 
                bety
              </span>
               
              <br />
              <strong
                className="score-modal "
              >
                 
                0
              </strong>
            </div>
          </div>
        </div>
        <div
          className="btns-block4"
        >
          <button
            className="btn-primary btn-color-1 btn-size-2 btn-orientation-auto "
            onClick={[Function]}
            type="button"
          >
            Cerrar
          </button>
        </div>
      </div>
    </div>,
    <div
      className="screen  "
    />,
  ]
</main>
`;
